src/libs/Navigation/AppNavigator/ModalStackNavigators.tsx

```diff
@@ -45,38 +46,46 @@ type Screens = Partial<Record<Screen, () => React.ComponentType>>;
  * @param getScreenOptions optional function that returns the screen options, override the default options
  */
-function createModalStackNavigator<TStackParams extends ParamListBase>(screens: Screens, getScreenOptions?: (styles: ThemeStyles) => StackNavigationOptions): React.ComponentType {
-    const ModalStackNavigator = createPlatformStackNavigator<TStackParams>();
-
-    function ModalStack() {
-        const styles = useThemeStyles();
-
-        const defaultSubRouteOptions = useMemo(
-            (): StackNavigationOptions => ({
-                ...subRouteOptions,
-                cardStyle: styles.navigationScreenCardStyle,
-            }),
-            [styles],
-        );
-
-        return (
-            <ModalStackNavigator.Navigator screenOptions={getScreenOptions?.(styles) ?? defaultSubRouteOptions}>
-                {Object.keys(screens as Required<Screens>).map((name) => (
-                    <ModalStackNavigator.Screen
-                        key={name}
-                        name={name}
-                        getComponent={(screens as Required<Screens>)[name as Screen]}
-                    />
-                ))}
-            </ModalStackNavigator.Navigator>
-        );
-    }
-
-    ModalStack.displayName = 'ModalStack';
-
-    return ModalStack;
+function createModalStackNavigatorFactory(factory: typeof createPlatformStackNavigator) {
+    return function createNestedModalStackNavigator<TStackParams extends ParamListBase>(
+        screens: Screens,
+        getScreenOptions?: (styles: ThemeStyles) => StackNavigationOptions,
+    ): React.ComponentType {
+        const ModalStackNavigator = factory<TStackParams>();
+
+        function ModalStack() {
+            const styles = useThemeStyles();
+
+            const defaultSubRouteOptions = useMemo(
+                (): StackNavigationOptions => ({
+                    ...subRouteOptions,
+                    cardStyle: styles.navigationScreenCardStyle,
+                }),
+                [styles],
+            );
+
+            return (
+                <ModalStackNavigator.Navigator screenOptions={getScreenOptions?.(styles) ?? defaultSubRouteOptions}>
+                    {Object.keys(screens as Required<Screens>).map((name) => (
+                        <ModalStackNavigator.Screen
+                            key={name}
+                            name={name}
+                            getComponent={(screens as Required<Screens>)[name as Screen]}
+                        />
+                    ))}
+                </ModalStackNavigator.Navigator>
+            );
+        }
+
+        ModalStack.displayName = 'ModalStack';
+
+        return ModalStack;
+    };
 }
 
+const createModalStackNavigator = createModalStackNavigatorFactory(createPlatformStackNavigator);
+const createJSModalStackNavigator = createModalStackNavigatorFactory(createStackNavigator);
```

Comment:
Can we just add a comment saying which one is preferred to use and why we have the JS one?